app/controllers/settings/email-preferences.js

```diff
@@ -0,0 +1,27 @@
+import Ember from 'ember';
+
+const { Controller } = Ember;
+
+export default Controller.extend({
+  actions: {
+    savePreference(notificationType, setting) {
+      this.get('store').findRecord('email-notification', setting.get('id'))
```

Comment:
Why all this getting the record again and needlessly changing property when the property is already changed and the record is already available at `setting` ?